### ISSUE TITLE
Conan omits empty directories when compressing a package

### ISSUE THREAD

**Issue by kdsx** [conan-io/conan#8013]
When Conan compresses a built package before upload, it omits all empty directories of the package. In some cases it can cause issue.

Example:
```py
from conans import ConanFile
import os


class EmptyDirTestConan(ConanFile):
    name = "empty-dir-test"
    version = "1.0"
    settings = None

    def build(self):
        os.makedirs(os.path.join(self.package_folder, "test", "empty_dir"))
        os.symlink("empty_dir", os.path.join(self.package_folder, "test", "link"))
```

We use two repositories for Conan packages (development and production). Packages are proposed from dev to prod when ready. Proposal is done via a sequence of `conan download` and `conan upload`. And here we face the issue. If a package contains an empty directory and a symlink which points to that directory, then we can't transfer it from one repo to another. When Conan uploads the package to a dev repo, it creates an archive without empty directories, thus symlinks became broken. After we do a transfer (download/upload), and Conan complains that the package has broken symlinks. I know that this check could be disabled, but this is not a proper solution.

### Environment Details (include every applicable attribute)
  * Operating System+version: Arch Linux (but it also reproduces on other distros)
  * Conan version: 1.31.0
  * Python version: 3.8.6

**Comment by jgsogo** [comment]
Using Conan 1.31, the _promotion_ from one repo to another using ``conan download`` + ``conan upload`` command shouldn't modify packages at all. The compressed files are preserved locally and the same binary that is downloaded from one repository is uploaded to the next one.

Regarding symlinks and empty folders: we have faced many issues over the past months and the proposal for **Conan 2.0 will probably be to compress the package folder as it is**, without checking symlinks or folders at all. There is a PR testing it is possible with many different folders layout (https://github.com/conan-io/conan/pull/7760), this will likely be the starting point for it. Before Conan 2.0 we cannot change this default behavior without breaking existing users.

**Comment by kdsx** [comment]
> Using Conan 1.31, the promotion from one repo to another using conan download + conan upload command shouldn't modify packages at all. The compressed files are preserved locally and the same binary that is downloaded from one repository is uploaded to the next one.

Sure. The issue is caused by compression. The promotion procedure is just an example of what could be broken by this issue.

**Comment by memsharded** [comment]
This was fixed in Conan 2.5, closing
